### PR TITLE
Extend build-infra to automate the creation of sysbox-in-docker container images

### PR DESCRIPTION
**Pull request by rodnymolina**
Our goal here is to automate the creation of docker images that contain an entire Sysbox sandbox. By doing this, users can easily interact with Sysbox runtime without needing to install it in their host machines.

```
$ make sysbox-in-docker

Usage:
  make sysbox-in-docker <distro-release>

  help                       Show supported docker images
  centos-8                   Build CentOS-8 docker image
  debian-buster              Build Debian-Buster docker image
  debian-bullseye            Build Debian-Bullseye docker image
  fedora-31                  Build Fedora-31 docker image
  fedora-32                  Build Fedora-32 docker image
  ubuntu-bionic              Build Ubuntu-Bionic docker image
  ubuntu-focal               Build Ubuntu-Focal docker image
```

Signed-off-by: Rodny Molina <rmolina@nestybox.com>

### CODE REVIEW

File: sysbox-in-docker/Dockerfile.debian-bullseye [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#
+# Sysbox-In-Docker Container Dockerfile (Debian-Bullseye image)
+#
+# This Dockerfile creates the sysbox-in-docker container image, which holds
+# all Sysbox binaries and its dependencies. The goal is to allow users to run
+# an entire Sysbox sandbox within a container.
+#
+# Notice that the image must be run as a privileged container, and a few resources
```
Comment (ctalledo) [inline sysbox-in-docker/Dockerfile.debian-bullseye:8]: I would reword this as:

NOTE: Sysbox is a container runtime and thus needs host root privileges. As a result, this image must be run as a privileged container ...  in inner containers. Note that within the privileged container, inner containers launched with Docker + Sysbox will be strongly isolated from the host by Sysbox (e.g., via the Linux user-namespace).

File: sysbox-in-docker/Dockerfile.debian-bullseye [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# Instructions:
+#
+# * Image creation:
+#
+#   $ make sysbox-in-docker debian-bullseye
+#
+# * Container creation:
+#
+# docker run -it --privileged --rm --hostname sysbox-in-docker --name sysbox-in-docker \
+#                -v /var/tmp/sysbox-test-var-lib-docker:/var/lib/docker \
```
Comment (ctalledo) [inline sysbox-in-docker/Dockerfile.debian-bullseye:21]: Remove reference to "sysbox-test-var-lib-docker" (it's a sysbox test dir). Use something like `-v some-vol:/var/lib/docker`

File: sysbox-in-docker/Dockerfile.debian-bullseye [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#                -v /var/tmp/sysbox-test-var-lib-sysbox:/var/lib/sysbox \
+#                -v /var/tmp/sysbox-test-scratch:/mnt/scratch \
```
Comment (ctalledo) [inline sysbox-in-docker/Dockerfile.debian-bullseye:23]: Don't need the scratch mount.

File: sysbox-in-docker/Makefile [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	@awk 'BEGIN {FS = ":.*##"; printf "\n\033[1mUsage:\n  \
+	make sysbox-in-docker \033[36m<distro-release>\033[0m\n\n"} \
+	/^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-25s\033[0m %s\n", $$1, $$2 } /^##@/ \
+	{ printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+
+LAUNCH_INSTRUCTION := docker run -it --privileged --rm --hostname sysbox-in-docker --name sysbox-in-docker \
+                -v /var/tmp/sysbox-test-var-lib-docker:/var/lib/docker \
+                -v /var/tmp/sysbox-test-var-lib-sysbox:/var/lib/sysbox \
+                -v /var/tmp/sysbox-test-scratch:/mnt/scratch \
```
Comment (ctalledo) [inline sysbox-in-docker/Makefile:37]: Don't need this.

File: sysbox-in-docker/Makefile [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+help: ## Show supported docker images
+	@awk 'BEGIN {FS = ":.*##"; printf "\n\033[1mUsage:\n  \
+	make sysbox-in-docker \033[36m<distro-release>\033[0m\n\n"} \
+	/^[a-zA-Z0-9_-]+:.*?##/ { printf "  \033[36m%-25s\033[0m %s\n", $$1, $$2 } /^##@/ \
+	{ printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+
+LAUNCH_INSTRUCTION := docker run -it --privileged --rm --hostname sysbox-in-docker --name sysbox-in-docker \
+                -v /var/tmp/sysbox-test-var-lib-docker:/var/lib/docker \
```
Comment (ctalledo) [inline sysbox-in-docker/Makefile:35]: Use other volumes (not the sysbox-test-var-lib-docker volume which is reserved for the sysbox test env).

File: sysbox-in-docker/sysbox [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+    if [ $? -ne 0 ]; then
+      printf "\dockerd failed to start.\n"
+      exit 1
+    fi
+  fi
+}
+
+
+function main() {
+
```
Comment (ctalledo) [inline sysbox-in-docker/sysbox:337]: remove empty line.

**Review by ctalledo — CHANGES_REQUESTED** [review]
Looks good, a few suggestions.

File: sysbox-in-docker/Makefile (hunk 5 above)
Comment (rodnymolina) [reply sysbox-in-docker/Makefile:35]: Done.

File: sysbox-in-docker/Makefile (hunk 4 above)
Comment (rodnymolina) [reply sysbox-in-docker/Makefile:37]: Done.

File: sysbox-in-docker/Dockerfile.debian-bullseye (hunk 3 above)
Comment (rodnymolina) [reply sysbox-in-docker/Dockerfile.debian-bullseye:23]: Done.

File: sysbox-in-docker/Dockerfile.debian-bullseye (hunk 2 above)
Comment (rodnymolina) [reply sysbox-in-docker/Dockerfile.debian-bullseye:21]: Done.

File: sysbox-in-docker/Dockerfile.debian-bullseye (hunk 1 above)
Comment (rodnymolina) [reply sysbox-in-docker/Dockerfile.debian-bullseye:8]: Done.

File: sysbox-in-docker/sysbox (hunk 6 above)
Comment (rodnymolina) [reply sysbox-in-docker/sysbox:337]: why?

**Review by ctalledo — APPROVED** [review]
